### ISSUE TITLE
memory consumption

### ISSUE THREAD

**Issue by treenerd** [kartoza/docker-mapproxy#46]
Hi,

First I would thank you for sharing this project.
I tried to use the container.
So I mapped the mapproxy cache directory and the config directory to an nfs share.
The lock files are written into dedicated directories under /tmp/locks and /tmp/tile_locks, like suggested in the mapproxy documentation.
It works fine at a first look.

But when I use the demo map, I saw the memory consumption of the container growing very fast. 
And it will not shrink anymore. Never saw this in a VM where I use mapproxy without docker.
I wonder if this is related to the docker and how it deals with a lot of small files created in sense of memory mappings.
Or if this is an issue of my configuration. 
I tested it also with the mapproxy demo map and saw the same "issue".
So I created my own container and tested with differend versions of mapproxy. 1.13.2 - 1.15
And saw the same issues.
At the end it will be an OOM Kill and the container will restart, but this is not nice in a production setup.
Did you see similar behavior, or is it an issue that just I have.
Would this stop with an S3 storage setup?

Thank you

**Comment by NyakudyaA** [comment]
Thanks for reporting this. Do you reproduce this with the sample docker-compose files here or your own custom examples? We would need to profile this, see the bottleneck, and fix it. Are your layers coming from a remote DB etc?

**Comment by treenerd** [comment]
Thank you for the fast response.

I work with k3s kubernetes, and docker, but I wrote my own deployment scripts.
Just to be sure I will test your docker compose scripts, do some more tests, and write it down a summary.
Could be it will take some time.
Hope I can do it during the weekend.

**Comment by treenerd** [comment]
Hi;

Had some time to investigate.

We figured out that there was an issue because of a setting in the mapproxy configuration file which caused in our case to much memory preasure when the memory is 90% full.
Seems like it is related to the `request_format: image/tiff` setting we used. 
At the end it causes an kill at the Docker Host: 

```
Sep 26 11:41:14 ***** kernel: pip invoked oom-killer: gfp_mask=0x6000c0(GFP_KERNEL), order=0, oom_score_adj=0
Sep 26 11:41:14 ***** kernel: CPU: 3 PID: 3657248 Comm: pip Not tainted 4.18.0-372.9.1.el8.x86_64 #1

```

We are not sure if it is related to the python-pillow version used inside the container. We will have to test. But as time is limited, we will see when we can do it.

Best Regards
